### PR TITLE
Added a fix and test for the removed-extension scenario for the Get-AzureDiskEncryptionStatus cmdlet

### PR DESCRIPTION
**Pull request by vermashi**
Fixed a bug where the correct status was not reported on native disk VMs when extension was removed.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/VirtualMachineExtensionTests.cs [hunk 1]
```diff
@@ -136,6 +136,18 @@ public void TestAzureDiskEncryptionExtensionSinglePass()
             ComputeTestController.NewInstance.RunPsTest(_logger, "Test-AzureDiskEncryptionExtensionSinglePass");
         }
 
+#if NETSTANDARD
+        [Fact(Skip = "Updated Storage, needs re-recorded")]
+        [Trait(Category.RunType, Category.DesktopOnly)]
+#else
+        [Fact(Skip = "TODO: only works for live mode due to key vault dependency")]
```
Comment (markcowl) [inline src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/VirtualMachineExtensionTests.cs]: Why would it only work for live mode?  KeyVault is just as mockable as network is.

File: src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs [hunk 2]
```diff
@@ -476,6 +477,18 @@ private void validateRetreivedVirtualMachine(VirtualMachine vm)
             }
         }
 
+        private bool isVMRunning (VirtualMachine vm)
+        {
+            string lastCode = "";
+            foreach (InstanceViewStatus ivs in vm.InstanceView.Statuses)
```
Comment (markcowl) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs]: Do you know that instanceView is non-null in this case?

File: src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs [hunk 3]
```diff
@@ -476,6 +477,18 @@ private void validateRetreivedVirtualMachine(VirtualMachine vm)
...
+            {
+                if (ivs.Code.StartsWith("PowerState/"))
```
Comment (markcowl) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs]: Do you know that ivs.Code is non-null in this case?

File: src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs [hunk 4]
```diff
@@ -495,7 +508,7 @@ private AzureDiskEncryptionStatusContext getStatusDualPass(VirtualMachine vm)
                     };
                     break;
                 case OperatingSystemTypes.Linux:
-                    if (!IsExtensionInstalled(vm))
+                    if (!this.IsExtensionInstalled(vm) && this.isVMRunning(vm))
```
Comment (markcowl) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs]: You don't need the ```this.```

File: src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs [hunk 5]
```diff
@@ -560,7 +573,7 @@ private AzureDiskEncryptionStatusContext getStatusSinglePass(VirtualMachine vm)
             AzureDiskEncryptionStatusContext status = this.GetStatusFromInstanceView(vm);
 
             // Get Data Disk status from extension for Native Disk VMs
-            if (isNativeDiskVM(vm))
+            if ( status.DataVolumesEncrypted != EncryptionStatus.NoDiskFound && isNativeDiskVM(vm))
```
Comment (markcowl) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs]: GetStatusFromInstanceView can return null, so this should likely be ```status?.DataVolumesEncrypted```

Comment (vermashi) [reply src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs]: GetStatusFromInstanceView never returns null.

**Comment by vermashi** [comment]
@markcowl  Any further changes needed?

**Pull request merged**